modules/aws/volume-access/main.tf

```diff
@@ -0,0 +1,138 @@
+      },
+      {
+        provider : "${value}",
+        test : "StringLike",
+        variable : "${value}:sub",
+        values : [format("system:serviceaccount:%s:*", var.external_id)]
+      }
+    ]
+  ]
+}
+
+resource "aws_iam_openid_connect_provider" "streamnative_oidc_providers" {
+  count          = var.init_oidc_providers ? length(local.oidc_providers) : 0
+  url            = "https://${local.oidc_providers[count.index]}"
+  client_id_list = ["sts.amazonaws.com"]
+  tags           = local.tag_set
+}
+
+data "aws_iam_policy_document" "streamnative_management_access" {
+  statement {
+    sid     = "AllowStreamNativeControlPlaneAccess"
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "AWS"
+      identifiers = var.streamnative_vendor_access_role_arns
+    }
+    dynamic "condition" {
+      for_each = local.assume_conditions
+      content {
+        test     = condition.value["test"]
+        values   = condition.value["values"]
+        variable = condition.value["variable"]
+      }
+    }
+  }
+
+  dynamic "statement" {
+    for_each = local.conditions
+    content {
+      effect  = "Allow"
+      actions = ["sts:AssumeRoleWithWebIdentity"]
+
+      principals {
+        type        = "Federated"
+        identifiers = [for provider in local.oidc_providers : "arn:aws:iam::${local.account_id}:oidc-provider/${provider}" if "${provider}" == statement.value[0].provider]
+      }
+
+      dynamic "condition" {
+        for_each = toset(statement.value)
+        content {
+          test     = condition.value["test"]
+          values   = condition.value["values"]
+          variable = condition.value["variable"]
+        }
+      }
+    }
+  }
+}
+
+######
+#-- Create the IAM role for the the StreamNative Cloud data access to s3 bucket
+######
+resource "aws_iam_policy" "access_bucket_role" {
+  name        = "sn-${var.external_id}-${var.bucket}-${var.path}"
+  description = "This policy sets the limits for the access s3 bucket for StreamNative's vendor access."
+  path        = "/StreamNative/"
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "s3:ListBucket"
+        ],
+        "Resource" : [
+          "arn:aws:s3:::${var.bucket}"
+        ]
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:DeleteObject"
+        ],
+        "Resource" : [
+          "arn:aws:s3:::${var.bucket}/${var.path}/*"
+        ]
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "s3:PutLifecycleConfiguration",
+          "s3:GetLifecycleConfiguration"
+        ],
+        "Resource" : [
+          "arn:aws:s3:::${var.bucket}/${var.path}"
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role" "access_bucket_role" {
+  name                 = "sn-${var.external_id}-${var.bucket}-${var.path}"
```

Comment:
The S3 path can contain multi path separator, e.g, `s3://test-snowflake/it_iceberg/data/streaming_ingest/GBQMaKNHsA/a1/`

Reply:
And be noted that the IAM Role name length is restricted to up to 64 characters.

Reply:
Fixed, thanks